lib/Control.js

```diff
@@ -0,0 +1,231 @@
+    }
+
+    function parsePrepared(headers, content) {
+        headers = parseHeaders(headers);
+        var contentLength = headers["content-length"];
+        return new PreparedEvent(parseContent(contentLength, content));
+    }
+
+    function parseStarted() {
+        return new StartedEvent();
+    }
+
+    var _this = this;
+    var buf = "";
+
+    this.connection.on('data', function (data) {
+        buf += data;
+        data = buf;
+        try {
+            do {
+                var eventTypeTerminator = data.indexOf("\n");
+                var headerTerminator = data.indexOf("\n\n");
+                if (headerTerminator < eventTypeTerminator || eventTypeTerminator <= 0) {
+                    // Not enough buffered to read from
+                    return;
+                } else {
+                    buf = buf.substring(headerTerminator + 2, buf.length);
+                }
+                var contentTerminator = data.length;
+                var eventType = data.substr(0, eventTypeTerminator);
+                var headers = data.substr(eventTypeTerminator + 1, headerTerminator);
+                var content = data.substr(headerTerminator + 2, contentTerminator);
+                var event;
+                //console.log("Received event " + eventType + ", with callbacks: " + _this.eventCallbacks[eventType]);
+                switch (eventType) {
+                    case "ERROR":
+                        event = parseError(headers, content);
+                        buf = buf.substring(event.getDescription().length, buf.length);
+                        break;
+                    case "FINISHED":
+                        event = parseFinished(headers, content);
+                        buf = buf.substring(event.getScript().length, buf.length);
+                        break;
+                    case "NOTIFIED":
+                        event = parseNotified(headers);
+                        break;
+                    case "PREPARED":
+                        event = parsePrepared(headers, content);
+                        buf = buf.substring(event.getScript().length, buf.length);
+                        break;
+                    case "STARTED":
+                        event = parseStarted();
+                        break;
+                    default:
+                        throw ("Unrecognized event: " + eventType);
+                }
+                if (!_this.eventCallbacks[eventType]) {
+                    _this.queuedEvents.push(event);
+                } else {
+                    for (var i = 0; i < _this.eventCallbacks[eventType].length; i++) {
+                        _this.eventCallbacks[eventType][i](event);
+                    }
+                }
+                data = buf;
+            } while (buf.length > 0);
+        } catch (err) {
+            if (err === "NOT_ENOUGH_BUFFERED") {
+                // This is an out for not having fully read tcp event
```

Comment:
It's not an error per say because the application will recover, and occasional tcp fragmentation is expected. It would be nice to put it at "trace" level in npm, but as far as I understand there is not an equivalent logger for npm, judging by [console](https://nodejs.org/api/console.html).  Though perhaps there is some other utility I should be using..